src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageElementDescriptionProvider.java

```diff
@@ -8,25 +8,79 @@
+import com.intellij.usageView.UsageViewShortNameLocation;
+import com.intellij.usageView.UsageViewTypeLocation;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
 import com.redhat.devtools.lsp4ij.LanguageServerBundle;
+import com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokenDocumentationProvider;
+import com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokenPsiElement;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
- * {@link ElementDescriptionProvider} implementation used to show "LSP Symbol" as target in the "Find Usages" view.
+ * {@link ElementDescriptionProvider} implementation used to show information about the target in the "Find Usages" view.
  */
 public class LSPUsageElementDescriptionProvider implements ElementDescriptionProvider {
     @Override
-    public @Nullable @NlsSafe String getElementDescription(@NotNull PsiElement element, @NotNull ElementDescriptionLocation location) {
-        if (element instanceof LSPUsageTriggeredPsiElement) {
-            return LanguageServerBundle.message("usage.description");
+    @Nullable
+    @NlsSafe
+    public String getElementDescription(@NotNull PsiElement element, @NotNull ElementDescriptionLocation location) {
+        // For usages
+        if (element instanceof LSPUsageTriggeredPsiElement usageTriggeredPsiElement) {
+            // Element type
+            if (location instanceof UsageViewTypeLocation) {
+                // See if we can get a better type
+                if (usageTriggeredPsiElement.getRealElement() instanceof LSPSemanticTokenPsiElement semanticTokenElement) {
+                    String tokenType = semanticTokenElement.getType();
+                    if (StringUtil.isNotEmpty(tokenType)) {
+                        return StringUtil.join(NameUtil.splitNameIntoWords(tokenType), String::toLowerCase, " ");
+                    }
+                }
+
+                // If we couldn't derive a type, return an empty string to avoid using the element's toString()
+                return "";
+            }
+
+            // Element name
+            else if (location instanceof UsageViewLongNameLocation) {
+                // See if we can get a better description
+                if (usageTriggeredPsiElement.getRealElement() instanceof LSPSemanticTokenPsiElement semanticTokenElement) {
+                    return semanticTokenElement.getName();
+                }
+
+                // Otherwise try to get the current word
+                PsiFile file = element.getContainingFile();
+                Document document = LSPIJUtils.getDocument(element);
+                TextRange wordRange = (file != null) && (document != null) ?
+                        LSPIJUtils.getWordRangeAt(document, file, element.getTextOffset()) :
+                        null;
+                if (wordRange != null) {
+                    return document.getText(wordRange);

```

Comment:
If we couldn't get a better name that way, try to use the text of the word that was used for finding usages.